cardano-node/src/Cardano/Node/Tracing/Tracers/P2P.hs

```diff
@@ -0,0 +1,1375 @@
+             , "targetActive" .= tActive
+             , "actualActive" .= aActive
+             , "peer" .= toJSON p
+             , "reason" .= show err
+             ]
+  forMachine _dtal (TracePromoteWarmDone tActive aActive p) =
+    mkObject [ "kind" .= String "PromoteWarmDone"
+             , "targetActive" .= tActive
+             , "actualActive" .= aActive
+             , "peer" .= toJSON p
+             ]
+  forMachine _dtal (TraceDemoteWarmPeers tEst aEst sp) =
+    mkObject [ "kind" .= String "DemoteWarmPeers"
+             , "targetEstablished" .= tEst
+             , "actualEstablished" .= aEst
+             , "selectedPeers" .= toJSONList (toList sp)
+             ]
+  forMachine _dtal (TraceDemoteWarmFailed tEst aEst p err) =
+    mkObject [ "kind" .= String "DemoteWarmFailed"
+             , "targetEstablished" .= tEst
+             , "actualEstablished" .= aEst
+             , "peer" .= toJSON p
+             , "reason" .= show err
+             ]
+  forMachine _dtal (TraceDemoteWarmDone tEst aEst p) =
+    mkObject [ "kind" .= String "DemoteWarmDone"
+             , "targetEstablished" .= tEst
+             , "actualEstablished" .= aEst
+             , "peer" .= toJSON p
+             ]
+  forMachine _dtal (TraceDemoteHotPeers tActive aActive sp) =
+    mkObject [ "kind" .= String "DemoteHotPeers"
+             , "targetActive" .= tActive
+             , "actualActive" .= aActive
+             , "selectedPeers" .= toJSONList (toList sp)
+             ]
+  forMachine _dtal (TraceDemoteLocalHotPeers taa sp) =
+    mkObject [ "kind" .= String "DemoteLocalHotPeers"
+             , "targetActualActive" .= toJSONList taa
+             , "selectedPeers" .= toJSONList (toList sp)
+             ]
+  forMachine _dtal (TraceDemoteHotFailed tActive aActive p err) =
+    mkObject [ "kind" .= String "DemoteHotFailed"
+             , "targetActive" .= tActive
+             , "actualActive" .= aActive
+             , "peer" .= toJSON p
+             , "reason" .= show err
+             ]
+  forMachine _dtal (TraceDemoteHotDone tActive aActive p) =
+    mkObject [ "kind" .= String "DemoteHotDone"
+             , "targetActive" .= tActive
+             , "actualActive" .= aActive
+             , "peer" .= toJSON p
+             ]
+  forMachine _dtal (TraceDemoteAsynchronous msp) =
+    mkObject [ "kind" .= String "DemoteAsynchronous"
+             , "state" .= toJSON msp
+             ]
+  forMachine _dtal TraceGovernorWakeup =
+    mkObject [ "kind" .= String "GovernorWakeup"
+             ]
+  forMachine _dtal (TraceChurnWait dt) =
+    mkObject [ "kind" .= String "ChurnWait"
+             , "diffTime" .= toJSON dt
+             ]
+  forMachine _dtal (TraceChurnMode c) =
+    mkObject [ "kind" .= String "ChurnMode"
+             , "event" .= show c ]
+  forHuman = pack . show
+
+docPeerSelection :: Documented (TracePeerSelection SockAddr)
+docPeerSelection = Documented [
+    DocMsg
+      (TraceLocalRootPeersChanged anyProto anyProto)
```

Comment:
Likewise, intentionally left empty?

Reply:
Yes it was intentionally, we can not provide documentation for all tracers.